src/Shared/ProblemDetails/HttpValidationProblemDetailsJsonConverter.cs

```diff
@@ -56,34 +41,88 @@ public static HttpValidationProblemDetails ReadProblemDetails(ref Utf8JsonReader
         }
 
         return problemDetails;
+
+        static void ReadErrors(ref Utf8JsonReader reader, IDictionary<string, string[]> errors)
+        {
+            if (!reader.Read())
+            {
+                throw new JsonException("Unexpected end when reading JSON.");
+            }
+
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.StartObject:
+                    while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+                    {
+                        var name = reader.GetString()!;
+
+                        if (!reader.Read())
+                        {
+                            throw new JsonException("Unexpected end when reading JSON.");
+                        }
+
+                        if (reader.TokenType == JsonTokenType.Null)
+                        {
+                            errors[name] = null!;
+                        }
+                        else
+                        {
+                            var values = new List<string>();
+                            while (reader.Read() && reader.TokenType != JsonTokenType.EndArray)
+                            {
+                                values.Add(reader.GetString()!);
+                            }
+                            errors[name] = values.ToArray();
```

Comment:
Are we concerned at all if the payload contains other information besides `string[]`? Would this code handle that case correctly?

Reply:
If someone sends an invalid value in the array then `GetString()` throws.